middlewares/osio/cache.go

```diff
@@ -43,14 +43,16 @@ func (r *ResolverPromise) Get() (interface{}, error) {
 
 	wg := sync.WaitGroup{}
 	wg.Add(1)
-
 	go func(wg *sync.WaitGroup, resolver Resolver) {
 		r.value, r.err = resolver()
 		wg.Done()
 
 	}(&wg, r.resolver)
 	wg.Wait()
-	r.resolved = true
+
+	if r.err == nil {
+		r.resolved = true
+	}
```

Comment:
In case when Resolver returns error .. resolved will NOT set to true .. so subsequent call to promise.Get() will attempt to call Resolver once again.
There are two reason.
first - At T1 time, resolver returning error (which is temporary, like due to tenant_service unavailable, it can't get required info and return HTTP 503).  At T2 time (next call), resolver returning value as tenant_service is available.
second - for v2, when w=false, At T1 time, this will return HTTP 404 as given namespace_type was not created. At T2 time (next call), it will return namespace as it was already created by now.